spec/lib/rbac/filterer_spec.rb

```diff
@@ -1286,56 +1287,143 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
         end
 
-        it "returns only the current user" do
-          get_rbac_results_for_and_expect_objects(User, [user])
+        context "and the current_group is the original regular group" do
+          before { user.current_group = group }
+
+          it "returns users from all of the current user's groups" do
+            get_rbac_results_for_and_expect_objects(User, [user, other_user])
+          end
+
+          it "returns user's groups" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, [group, other_group])
+          end
         end
 
-        it "returns only the current group" do
-          get_rbac_results_for_and_expect_objects(MiqGroup, [user.current_group])
+        context "and the current_group is another regular group" do
+          before { user.current_group = other_group }
+
+          it "returns users from all of the current user's groups" do
+            get_rbac_results_for_and_expect_objects(User, [user, other_user])
+          end
+
+          it "returns user's groups" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, [group, other_group])
+          end
         end
       end
 
-      context 'with EvmRole-tenant_administrator' do
-        let(:rbac_tenant) do
-          FactoryBot.create(:miq_product_feature, :identifier => MiqProductFeature::TENANT_ADMIN_FEATURE)
+      context "where the user is in a group that is a super_administrator group" do
+        before do
+          admin_user # ensure we create another user as an admin user
+          other_user # ensure we create another user in another regular group
+
+          user.miq_groups << admin_group
+          user.miq_groups << other_group
         end
 
-        let(:tenant_administrator_user_role) do
-          FactoryBot.create(:miq_user_role, :name => MiqUserRole::DEFAULT_TENANT_ROLE_NAME, :miq_product_features => [rbac_tenant])
+        context "and the current_group is not the super_administrator group" do
+          before { user.current_group = group }
+
+          it "returns users from all of the current user's groups except admins" do
+            get_rbac_results_for_and_expect_objects(User, [user, other_user])
+          end
+
+          it "returns user's groups except admins" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, [group, other_group])
+          end
         end
 
-        let!(:super_administrator_user_role) do
-          FactoryBot.create(:miq_user_role, :role => "super_administrator")
+        context "and the current_group is the super_administrator group" do
+          before { user.current_group = admin_group }
+
+          it "returns all users" do
+            get_rbac_results_for_and_expect_objects(User, User.all.to_a)
+          end
+
+          it "returns user's groups" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, MiqGroup.all.to_a)
+          end
         end
+      end
+
+      context "with a self_service user" do
```

Comment:
this is a regular self service user and not a limited self service user. right?

Reply:
That's correct